### ISSUE TITLE
BSRNN with `causal=True` is not causal

### ISSUE THREAD

**Issue by philgzl** [espnet/espnet#5900]
It seems to me that BSRNN with `causal=True` is not causal. Forward passing a tensor with NaNs in the last time frame results in a tensor with NaNs only.

```python
import torch

from espnet2.enh.layers.bsrnn import BSRNN

net = BSRNN(causal=True)
net.eval()

x = torch.randn(1, 1000, 481, 2)
x[:, -1, :, :] = float('nan')
x = net(x)
assert not x.isnan().all()  # fails
```

The first non-causal operation seems to be the normalization layers in the band-split module, which calculate statistics over time frames.

**Comment by sw005320** [comment]
Thanks for pointing out the issue.
@Emrys365, can you check this issue?

**Comment by Emrys365** [comment]
Hi @philgzl, to make the model fully causal, you will also need to set the `norm_type` to either "cLN" or "cfLN" in addition to `causal=True`.

**Comment by philgzl** [comment]
Thanks @Emrys365. Can you confirm that the configurations marked as causal in [your paper](https://arxiv.org/abs/2406.04269) have this modification? The `norm_type` option does not seem to be specified in the configuration files provided in [Emrys365/se-scaling](https://github.com/Emrys365/se-scaling).

**Comment by Emrys365** [comment]
No, the [paper](https://arxiv.org/abs/2406.04269) only employed the `causal=True` argument, so the models are not strictly causal (all operations but the normalization layers are causal).

**Comment by philgzl** [comment]
Cool, thanks for clarifying!
